### PR TITLE
New semantic analyzer: enable remaining tests and fix issues

### PR DESCRIPTION
**Pull request by JukkaL**
This enables incremental mode tests for the new semantic analyzer. Some
tests are skipped, mostly because of issues related to recursive type 
definitions and plugins.

Also fix some issues related to type aliases and recursive definitions.

Remove support for skipping test files when using the new semantic 
analyzer.

### CODE REVIEW

File: mypy/newsemanal/semanal.py [hunk 1]
```diff
@@ -2349,6 +2352,10 @@ def check_and_set_up_type_alias(self, s: AssignmentStmt) -> bool:
         if not self.can_be_type_alias(rvalue):
             return False
 
+        if existing and not isinstance(existing.node, (PlaceholderNode, TypeAlias)):
+            # Cannot redefine existing node as type alias.
+            return False
```
Comment (ilevkivskyi) [inline mypy/newsemanal/semanal.py]: OK, moving this upper makes sense.

File: mypy/newsemanal/semanal.py [hunk 2]
```diff
@@ -2390,21 +2397,28 @@ def check_and_set_up_type_alias(self, s: AssignmentStmt) -> bool:
...
-            if (isinstance(existing.node, PlaceholderNode) or
-                    isinstance(existing.node, TypeAlias) and existing.node.target != res):
-                self.progress = True
-                # We need to defer so that this change can get propagated to base classes.
-                self.defer()
+            # An alias gets updated.
+            if self.final_iteration:
+                # Error will be reported elsewhere, but we can't make progress during
+                # final iteration.
+                return True
```
Comment (ilevkivskyi) [inline mypy/newsemanal/semanal.py]: This part (check for final iteration) looks a bit suspicious. Could you please add a bit longer comment about what kind of situation may cause this, and what error will be reported (will this be just invalid type in assignment, I noticed a TODO about this)?

Comment (JukkaL) [reply mypy/newsemanal/semanal.py]: Added an explicit error message, since it wasn't clear if an error will be reported elsewhere.

**Comment by JukkaL** [comment]
Opening and closing to get CI builds to run again, as I suspect that Travis is flaky.

**Comment by JukkaL** [comment]
Merging even though Travis failed, since it looks like Travis is flaky.

**Pull request merged**